// cielo/dist/cielo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cielo = void 0;
const eletronic_transfer_1 = require("./class/eletronic-transfer");
const bank_slip_1 = require("./class/bank-slip");
const recurrent_1 = require("./class/recurrent");
const consult_1 = require("./class/consult");
const cards_1 = require("./class/cards");
const creditcard_1 = require("./class/creditcard");
const debit_card_1 = require("./class/debit-card");
class Cielo {
    constructor(constructor) {
        this.cartao = this.card;
        this.merchantId = constructor.merchantId;
        this.merchantKey = constructor.merchantKey;
        this.debug = constructor.debug || false;
        this.sandbox = constructor.sandbox || false;
        this.requestId = constructor.requestId || undefined;
        const [hostnameTransacao, hostnameQuery] = this.getHostnames(this.sandbox);
        const cieloTransactionInterface = {
            hostnameTransacao,
            hostnameQuery,
            merchantId: this.merchantId,
            merchantKey: this.merchantKey,
            requestId: this.requestId,
        };
        this.creditCard = new creditcard_1.CreditCard(cieloTransactionInterface);
        this.debitCard = new debit_card_1.DebitCard(cieloTransactionInterface);
        this.card = new cards_1.Card(cieloTransactionInterface);
        this.consult = new consult_1.Consult(cieloTransactionInterface);
        this.recurrent = new recurrent_1.Recurrent(cieloTransactionInterface);
        this.bankSlip = new bank_slip_1.BankSlip(cieloTransactionInterface);
        this.eletronicTransfer = new eletronic_transfer_1.EletronicTransfer(cieloTransactionInterface);
        this.consulta = this.consult;
        this.recorrencia = this.recurrent;
        this.boleto = this.bankSlip;
        this.transferenciaEletronica = this.eletronicTransfer;
    }
    getHostnames(sandbox) {
        if (sandbox) {
            return [
                'apisandbox.cieloecommerce.cielo.com.br',
                'apiquerysandbox.cieloecommerce.cielo.com.br'
            ];
        }
        else {
            return [
                'api.cieloecommerce.cielo.com.br',
                'apiquery.cieloecommerce.cielo.com.br',
            ];
        }
    }
}
exports.Cielo = Cielo;
