// cielo/dist/enums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumFraudAnalysisStatus = exports.EnumCartVelocityHedge = exports.EnumCartType = exports.EnumCartTimeHedge = exports.EnumCartPhoneHedge = exports.EnumCartObscenitiesHedge = exports.EnumCartNonSensicalHedge = exports.EnumCartHostHedge = exports.EnumCartCategory = exports.EnumTravelPassengersRating = exports.EnumTravelPassengersStatus = exports.EnumTravelJourneyType = exports.EnumShippingMethod = exports.EnumSequenceCriteria = exports.EnumTypeFlowAnalysisFraud = exports.EnumRecurrentPaymentUpdateInterval = exports.EnumRecurrentPaymentInterval = exports.EnumBrands = exports.EnumCardType = exports.EnumPaymentInterest = exports.EnumIdentityType = void 0;
var EnumIdentityType;
(function (EnumIdentityType) {
    EnumIdentityType["RG"] = "RG";
    EnumIdentityType["CPF"] = "CPF";
    EnumIdentityType["CNPJ"] = "CNPJ";
})(EnumIdentityType = exports.EnumIdentityType || (exports.EnumIdentityType = {}));
var EnumPaymentInterest;
(function (EnumPaymentInterest) {
    EnumPaymentInterest["MERCHANT"] = "ByMerchant";
    EnumPaymentInterest["ISSUER"] = "ByIssuer";
})(EnumPaymentInterest = exports.EnumPaymentInterest || (exports.EnumPaymentInterest = {}));
var EnumCardType;
(function (EnumCardType) {
    EnumCardType["CREDIT"] = "CreditCard";
    EnumCardType["DEBIT"] = "DebitCard";
})(EnumCardType = exports.EnumCardType || (exports.EnumCardType = {}));
var EnumBrands;
(function (EnumBrands) {
    EnumBrands["VISA"] = "VISA";
    EnumBrands["MASTER"] = "MASTER";
    EnumBrands["AMEX"] = "AMEX";
    EnumBrands["ELO"] = "ELO";
    EnumBrands["AURA"] = "AURA";
    EnumBrands["JCB"] = "JCB";
    EnumBrands["DINERS"] = "DINERS";
    EnumBrands["DISCOVERY"] = "DISCOVER";
    EnumBrands["DISCOVER"] = "DISCOVER";
    EnumBrands["HIPERCARD"] = "HIPERCARD";
})(EnumBrands = exports.EnumBrands || (exports.EnumBrands = {}));
var EnumRecurrentPaymentInterval;
(function (EnumRecurrentPaymentInterval) {
    EnumRecurrentPaymentInterval["MONTHLY"] = "MONTHLY";
    EnumRecurrentPaymentInterval["BIMONTHLY"] = "BIMONTHLY";
    EnumRecurrentPaymentInterval["QUARTERLY"] = "QUARTERLY";
    EnumRecurrentPaymentInterval["SEMIANNUAL"] = "SEMIANNUAL";
    EnumRecurrentPaymentInterval["ANNUAL"] = "ANNUAL";
})(EnumRecurrentPaymentInterval = exports.EnumRecurrentPaymentInterval || (exports.EnumRecurrentPaymentInterval = {}));
var EnumRecurrentPaymentUpdateInterval;
(function (EnumRecurrentPaymentUpdateInterval) {
    EnumRecurrentPaymentUpdateInterval[EnumRecurrentPaymentUpdateInterval["MONTHLY"] = 1] = "MONTHLY";
    EnumRecurrentPaymentUpdateInterval[EnumRecurrentPaymentUpdateInterval["BIMONTHLY"] = 2] = "BIMONTHLY";
    EnumRecurrentPaymentUpdateInterval[EnumRecurrentPaymentUpdateInterval["QUARTERLY"] = 3] = "QUARTERLY";
    EnumRecurrentPaymentUpdateInterval[EnumRecurrentPaymentUpdateInterval["SEMIANNUAL"] = 6] = "SEMIANNUAL";
    EnumRecurrentPaymentUpdateInterval[EnumRecurrentPaymentUpdateInterval["ANNUAL"] = 12] = "ANNUAL";
})(EnumRecurrentPaymentUpdateInterval = exports.EnumRecurrentPaymentUpdateInterval || (exports.EnumRecurrentPaymentUpdateInterval = {}));
var EnumTypeFlowAnalysisFraud;
(function (EnumTypeFlowAnalysisFraud) {
    EnumTypeFlowAnalysisFraud["AnalyseFirst"] = "AnalyseFirst";
    EnumTypeFlowAnalysisFraud["AuthorizeFirst"] = "AuthorizeFirst";
})(EnumTypeFlowAnalysisFraud = exports.EnumTypeFlowAnalysisFraud || (exports.EnumTypeFlowAnalysisFraud = {}));
var EnumSequenceCriteria;
(function (EnumSequenceCriteria) {
    EnumSequenceCriteria["OnSuccess"] = "OnSuccess";
    EnumSequenceCriteria["Always"] = "Always";
})(EnumSequenceCriteria = exports.EnumSequenceCriteria || (exports.EnumSequenceCriteria = {}));
var EnumShippingMethod;
(function (EnumShippingMethod) {
    EnumShippingMethod["SameDay"] = "SameDay";
    EnumShippingMethod["OneDay"] = "OneDay";
    EnumShippingMethod["TwoDay"] = "TwoDay";
    EnumShippingMethod["ThreeDay"] = "ThreeDay";
    EnumShippingMethod["LowCost"] = "LowCost";
    EnumShippingMethod["Pickup"] = "Pickup";
    EnumShippingMethod["Other"] = "Other";
    EnumShippingMethod["None"] = "None"; //	Sem meio de entrega, pois é um serviço ou assinatura
})(EnumShippingMethod = exports.EnumShippingMethod || (exports.EnumShippingMethod = {}));
var EnumTravelJourneyType;
(function (EnumTravelJourneyType) {
    EnumTravelJourneyType["OneWayTrip"] = "OneWayTrip";
    EnumTravelJourneyType["RoundTrip"] = "RoundTrip"; //	Viagem de ida e volta
})(EnumTravelJourneyType = exports.EnumTravelJourneyType || (exports.EnumTravelJourneyType = {}));
var EnumTravelPassengersStatus;
(function (EnumTravelPassengersStatus) {
    EnumTravelPassengersStatus["Standard"] = "Standard";
    EnumTravelPassengersStatus["Gold"] = "Gold";
    EnumTravelPassengersStatus["Platinum"] = "Platinum";
})(EnumTravelPassengersStatus = exports.EnumTravelPassengersStatus || (exports.EnumTravelPassengersStatus = {}));
var EnumTravelPassengersRating;
(function (EnumTravelPassengersRating) {
    EnumTravelPassengersRating["Adult"] = "Adult";
    EnumTravelPassengersRating["Child"] = "Child";
    EnumTravelPassengersRating["Infant"] = "Infant"; //	Infantil
})(EnumTravelPassengersRating = exports.EnumTravelPassengersRating || (exports.EnumTravelPassengersRating = {}));
var EnumCartCategory;
(function (EnumCartCategory) {
    EnumCartCategory["Yes"] = "Yes";
    EnumCartCategory["No"] = "No";
    EnumCartCategory["Off"] = "Off"; // Diferenças entre os endereços de cobrança e entrega não afetam a pontuação
})(EnumCartCategory = exports.EnumCartCategory || (exports.EnumCartCategory = {}));
var EnumCartHostHedge;
(function (EnumCartHostHedge) {
    EnumCartHostHedge["Low"] = "Low";
    EnumCartHostHedge["Normal"] = "Normal";
    EnumCartHostHedge["High"] = "High";
    EnumCartHostHedge["Off"] = "Off"; //	Não irá afetar o score da análise de fraude
})(EnumCartHostHedge = exports.EnumCartHostHedge || (exports.EnumCartHostHedge = {}));
var EnumCartNonSensicalHedge;
(function (EnumCartNonSensicalHedge) {
    EnumCartNonSensicalHedge["Low"] = "Low";
    EnumCartNonSensicalHedge["Normal"] = "Normal";
    EnumCartNonSensicalHedge["High"] = "High";
    EnumCartNonSensicalHedge["Off"] = "Off"; //	Não irá afetar o score da análise de fraude
})(EnumCartNonSensicalHedge = exports.EnumCartNonSensicalHedge || (exports.EnumCartNonSensicalHedge = {}));
var EnumCartObscenitiesHedge;
(function (EnumCartObscenitiesHedge) {
    EnumCartObscenitiesHedge["Low"] = "Low";
    EnumCartObscenitiesHedge["Normal"] = "Normal";
    EnumCartObscenitiesHedge["High"] = "High";
    EnumCartObscenitiesHedge["Off"] = "Off"; //	Não irá afetar o score da análise de fraude
})(EnumCartObscenitiesHedge = exports.EnumCartObscenitiesHedge || (exports.EnumCartObscenitiesHedge = {}));
var EnumCartPhoneHedge;
(function (EnumCartPhoneHedge) {
    EnumCartPhoneHedge["Low"] = "Low";
    EnumCartPhoneHedge["Normal"] = "Normal";
    EnumCartPhoneHedge["High"] = "High";
    EnumCartPhoneHedge["Off"] = "Off"; //	Não irá afetar o score da análise de fraude
})(EnumCartPhoneHedge = exports.EnumCartPhoneHedge || (exports.EnumCartPhoneHedge = {}));
var EnumCartTimeHedge;
(function (EnumCartTimeHedge) {
    EnumCartTimeHedge["Low"] = "Low";
    EnumCartTimeHedge["Normal"] = "Normal";
    EnumCartTimeHedge["High"] = "High";
    EnumCartTimeHedge["Off"] = "Off"; //	Não irá afetar o score da análise de fraude
})(EnumCartTimeHedge = exports.EnumCartTimeHedge || (exports.EnumCartTimeHedge = {}));
var EnumCartType;
(function (EnumCartType) {
    EnumCartType["AdultContent"] = "AdultContent";
    EnumCartType["Coupon"] = "Coupon";
    EnumCartType["Default"] = "Default";
    EnumCartType["EletronicGood"] = "EletronicGood";
    EnumCartType["EletronicSoftware"] = "EletronicSoftware";
    EnumCartType["GiftCertificate"] = "GiftCertificate";
    EnumCartType["HandlingOnly"] = "HandlingOnly";
    EnumCartType["Service"] = "Service";
    EnumCartType["ShippingAndHandling"] = "ShippingAndHandling";
    EnumCartType["ShippingOnly"] = "ShippingOnly";
    EnumCartType["Subscription"] = "Subscription";
})(EnumCartType = exports.EnumCartType || (exports.EnumCartType = {}));
var EnumCartVelocityHedge;
(function (EnumCartVelocityHedge) {
    EnumCartVelocityHedge["Low"] = "Low";
    EnumCartVelocityHedge["Normal"] = "Normal";
    EnumCartVelocityHedge["High"] = "High";
    EnumCartVelocityHedge["Off"] = "Off"; //	Não irá afetar o score da análise de fraude
})(EnumCartVelocityHedge = exports.EnumCartVelocityHedge || (exports.EnumCartVelocityHedge = {}));
var EnumFraudAnalysisStatus;
(function (EnumFraudAnalysisStatus) {
    EnumFraudAnalysisStatus[EnumFraudAnalysisStatus["Unknown"] = 0] = "Unknown";
    EnumFraudAnalysisStatus[EnumFraudAnalysisStatus["Accept"] = 1] = "Accept";
    EnumFraudAnalysisStatus[EnumFraudAnalysisStatus["Reject"] = 2] = "Reject";
    EnumFraudAnalysisStatus[EnumFraudAnalysisStatus["Review"] = 3] = "Review";
    EnumFraudAnalysisStatus[EnumFraudAnalysisStatus["Aborted"] = 4] = "Aborted";
    EnumFraudAnalysisStatus[EnumFraudAnalysisStatus["Unfinished"] = 5] = "Unfinished";
})(EnumFraudAnalysisStatus = exports.EnumFraudAnalysisStatus || (exports.EnumFraudAnalysisStatus = {}));
