// cielo/dist/class/consult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Consult = void 0;
const utils_1 = require("./utils");
class Consult {
    constructor(transaction) {
        this.cieloTransactionParams = transaction;
        this.util = new utils_1.Utils(transaction);
    }
    paymentId(params) {
        const options = {
            path: `/1/sales/${params.paymentId}`,
        };
        return this.util.get(options);
    }
    merchantOrderId(params) {
        const options = {
            path: `/1/sales?merchantOrderId=${params.merchantOrderId}`,
        };
        return this.util.get(options);
    }
    recurrent(params) {
        const options = {
            path: `/1/RecurrentPayment/${params.recurrentPaymentId}`,
        };
        return this.util.get(options);
    }
    bin(params) {
        const options = {
            path: `/1/cardBin/${params.cardBin}`,
        };
        return this.util.get(options);
    }
    cardtoken(params) {
        const options = {
            path: `/1/card/${params.cardToken}`,
        };
        return this.util.get(options);
    }
}
exports.Consult = Consult;
